Microsoft.Toolkit.Uwp.UI/ImageCache/ImageCache.cs

```diff
@@ -243,58 +237,47 @@ public static Task<BitmapImage> GetFromCacheAsync(Uri uri, bool throwOnError = f
+            var folder = await GetCacheFolderAsync();
+
+            var baseFile = await folder.TryGetItemAsync(key) as StorageFile;
+            if (await IsFileOutOfDate(baseFile, expirationDate))
+            {
+                baseFile = await folder.CreateFileAsync(key, CreationCollisionOption.ReplaceExisting);
+                try
                 {
-                    baseFile = await folder.CreateFileAsync(key, CreationCollisionOption.ReplaceExisting);
-                    try
+                    using (var webStream = await StreamHelper.GetHttpStreamAsync(uri))
                     {
-                        using (var webStream = await StreamHelper.GetHttpStreamAsync(uri))
+                        using (var reader = new DataReader(webStream))
                         {
-                            if (!isPreCache)
-                            {
-                                image = new BitmapImage();
-                                image.SetSource(webStream);
-                                webStream.Seek(0);
-                            }
-
-                            using (var reader = new DataReader(webStream))
-                            {
-                                await reader.LoadAsync((uint)webStream.Size);
-                                var buffer = new byte[(int)webStream.Size];
-                                reader.ReadBytes(buffer);
-                                await FileIO.WriteBytesAsync(baseFile, buffer);
-                            }
+                            await reader.LoadAsync((uint)webStream.Size);
+                            var buffer = new byte[(int)webStream.Size];
+                            reader.ReadBytes(buffer);
+                            await FileIO.WriteBytesAsync(baseFile, buffer);
                         }
                     }
-                    catch (Exception e)
-                    {
-                        await baseFile.DeleteAsync();
-                        throw e;
-                    }
                 }
-                else
+                catch (Exception)
                 {
-                    if (isPreCache)
-                    {
-                        return null;
-                    }
-
-                    using (var fileStream = await baseFile.OpenAsync(FileAccessMode.Read))
-                    {
-                        image = new BitmapImage();
-                        image.SetSource(fileStream);
-                    }
+                    await baseFile.DeleteAsync();
+                    throw; // rethrowing the exception changes the stack trace. just throw
                 }
+            }
 
-                if (MaxMemoryCacheSize > 0 && image != null)
-                {
-                    StoreToMemoryCache(key, image);
-                }
+            // the baseFile should be valid at this point and not null
+            using (var fileStream = await baseFile.OpenAsync(FileAccessMode.Read))
```

Comment:
@jurepurgar i have reworked it and PreCache and PreCache in memory have clear performance differences now